code/game/objects/items/stacks/stack.dm

```diff
@@ -177,11 +176,13 @@
 
 		src.produce_recipe(R, multiplier, usr)
 
-	if (src && usr.machine==src) //do not reopen closed window
-		spawn( 0 )
-			src.interact(usr)
-			return
-	return
+	if(src && usr.machine == src) //do not reopen closed window
+		do_nonblocking_interact(usr)
+
+/obj/item/stack/proc/do_nonblocking_interact(var/mob/user)
+	set waitfor = FALSE
+	sleep(0)
+	interact(user)
```

Comment:
Unless you are manually calling `Topic` somewhere (never do that), you should be able to just call `interact(user)` at the end. You aren't unblocking for anything in the rest of `Topic` because this is at the end of `Topic`.

`src` will never be null unless you manually null it or are doing `call` abuse. `usr.machine == src` is not a reliable check of anything, but I guess you can keep it if you like.